en/tutorials-and-examples/cms/articles-controller.rst

```diff
@@ -1,9 +1,132 @@
 CMS Tutorial - Creating the Articles Controller
 ###############################################
 
-* Create the Controller
-* Add a index & view action.
-* Add the index & view template
+With our model created, we need a controller for our articles. Controllers in
+CakePHP handle HTTP requests and execute business logic contained in model
+methods, to prepare the response. We'll place this new controller in a file
+called **ArticlesController.php** inside the **src/Controller** directory.
+Here's what the basic controller should look like::
+
+    // src/Controller/ArticlesController.php
+
+    namespace App\Controller;
+
+    class ArticlesController extends AppController
+    {
+    }
+
+Now, let's add an action to our controller. Actions are controller methods that
+have routes connected to them. For example, when a user requests
+**www.example.com/articles/index** (which is also the same as
+**www.example.com/articles**), CakePHP will call the ``index`` method of your
+``ArticlesController``. This method should query the model layer, and prepare
+a response (via the View and Template). The code for that action would look like
+this::
+
+    // src/Controller/ArticlesController.php
+
+    namespace App\Controller;
+
+    class ArticlesController extends AppController
+    {
+
+        public function index()
+        {
+            $articles = $this->Articles->find('all');
+            $this->set(compact('articles'));
+        }
+    }
+
+By defining function ``index()`` in our ``ArticlesController``, users can now
+access the logic there by requesting **www.example.com/articles/index**. Similarly,
+if we were to define a function called ``foobar()``, users would be able to
+access that at **www.example.com/articles/foobar**.
+
+.. warning::
+
+    You may be tempted to name your controllers and actions in a way that allows
```

Comment:
While I think it's good to have this warning, I feel it pulls away the concentration of a reader. Could we move this below the next  paragraph?